From 2b3e8411ffbc2cab018824edefcc067b6d1fc8f1 Mon Sep 17 00:00:00 2001
From: 0xD34D <clark@scheffsblend.com>
Date: Sat, 11 Nov 2023 06:54:42 +0000
Subject: [PATCH 04/43] Add missing get_pin_info method to stepper.py

---
 klippy/stepper.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/klippy/stepper.py b/klippy/stepper.py
index 56c8ec758..a6aa25eb7 100644
--- a/klippy/stepper.py
+++ b/klippy/stepper.py
@@ -58,6 +58,8 @@ class MCU_stepper:
         if short and self._name.startswith('stepper_'):
             return self._name[8:]
         return self._name
+    def get_pin_info(self):
+        return self._dir_pin, self._step_pin, self._invert_dir, self._invert_step
     def units_in_radians(self):
         # Returns true if distances are in radians instead of millimeters
         return self._units_in_radians
-- 
2.39.5 (Apple Git-154)

