From cdc11ba5002f1603f475de0213558f253a42cea1 Mon Sep 17 00:00:00 2001
From: 0xD34D <clark@scheffsblend.com>
Date: Tue, 14 Nov 2023 06:48:18 +0000
Subject: [PATCH 13/43] Probe sensor location twice using the bltouch

---
 klippy/extras/prtouch.py | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/klippy/extras/prtouch.py b/klippy/extras/prtouch.py
index 41182afa4..a90940e18 100644
--- a/klippy/extras/prtouch.py
+++ b/klippy/extras/prtouch.py
@@ -813,9 +813,11 @@ class PRTouchEndstopWrapper:
         probe_y = self.cfg.sensor_y - probe_y_offset
         self.pnt_msg("Checking z-position of probe (%.2f, %.2f)" % (probe_x, probe_y))
         self._move([probe_x, probe_y, self.cfg.bed_max_err + 1.], self.cfg.g29_xy_speed)
-        z_probe = self.obj.probe.run_probe(gcmd)
+        probe_gcmd = self.obj.gcode.create_gcode_command("PROBE", "PROBE", {'SAMPLES': '2'})
+        z_probe = self.obj.probe.run_probe(probe_gcmd)
+        self.pnt_msg('Z_PROBE: %.3f' % z_probe[2])
         z_offset = self.probe_z_offset() - z_probe[2]
-        self.pnt_msg('Z_OFFSET: %.2f' % z_offset)
+        self.pnt_msg('Z_OFFSET: %.3f' % z_offset)
         self.obj.probe.z_offset = -z_offset
         z_probe[2] = z_offset
         self.obj.probe.probe_calibrate_finalize(z_probe)
-- 
2.39.5 (Apple Git-154)

