From 5e44ef1e78ac1c9c106cad4ee5e1062e7d324101 Mon Sep 17 00:00:00 2001
From: SupremeCake17 <132528442+SupremeCake17@users.noreply.github.com>
Date: Thu, 30 May 2024 18:13:42 -0500
Subject: [PATCH 38/43] Fixed Gcode command sending in printerInterface

---
 klippy/extras/printerInterface.py | 38 ++++++++++++++++---------------
 1 file changed, 20 insertions(+), 18 deletions(-)

diff --git a/klippy/extras/printerInterface.py b/klippy/extras/printerInterface.py
index 3df5803a8..c15421d13 100644
--- a/klippy/extras/printerInterface.py
+++ b/klippy/extras/printerInterface.py
@@ -155,6 +155,7 @@ class PrinterData:
         self.reactor = self.printer.get_reactor()
         self._logging = config.getboolean("logging", False)
         self.gcode = self.printer.lookup_object("gcode")
+        self.fl = []
         self.status = None
  
 
@@ -217,9 +218,11 @@ class PrinterData:
     def GetFiles(self, refresh=False):
         sdcard = self.printer.lookup_object('virtual_sdcard')
         files = sdcard.get_file_list(True)
+        self.fl = []
         self.names = []
         for file, _ in files:
-            self.names.append(file)
+            self.fl.append(file)
+            self.names.append(file.split('/')[-1])
         return self.names
 
     def update_variable(self):
@@ -325,7 +328,7 @@ class PrinterData:
             )
         )
 
-    def SendGCode(self, Gcode):
+    def sendGCode(self, Gcode):
         gcode = self.printer.lookup_object('gcode')
         gcode._process_commands([Gcode])
 
@@ -354,47 +357,46 @@ class PrinterData:
         self.setExtTemp(exttemp)
 
     
-    def OpenAndPrintFile(self, filenum):
-        sdcard = self.printer.lookup_object('virtual_sdcard')
-        files = sdcard.cmd_SDCARD_PRINT_FILE(self.names[filenum])
+    def openAndPrintFile(self, filenum):
+        self.sendGCode('SDCARD_PRINT_FILE FILENAME="{}"'.format(self.fl[filenum]))
 
-    def SendGCode(self, Gcode):
+    def sendGCode(self, Gcode):
         self.gcode._process_commands([Gcode])
 
     def probe_calibrate(self):
-        self.SendGCode('G28') # home the printer
-        self.SendGCode('PRTOUCH_PROBE_OFFSET CLEAR_NOZZLE=0 APPLY_Z_ADJUST=1') # use the prtouch to find the z offset and apply it
+        self.sendGCode('G28') # home the printer
+        self.sendGCode('PRTOUCH_PROBE_OFFSET CLEAR_NOZZLE=0 APPLY_Z_ADJUST=1') # use the prtouch to find the z offset and apply it
 
     def resume_job(self):
-        self.SendGCode('RESUME') # resume the print
+        self.sendGCode('RESUME') # resume the print
 
     def pause_job(self):
-        self.SendGCode('PAUSE') # pause the print
+        self.sendGCode('PAUSE') # pause the print
 
     def cancel_job(self):
-        self.SendGCode('CANCEL_PRINT') # cancel the print
+        self.sendGCode('CANCEL_PRINT') # cancel the print
 
     def set_feedrate(self, value):
-        self.SendGCode('M220 S' + str(value)) # set the feedrate through the M220 gcode command
+        self.sendGCode('M220 S' + str(value)) # set the feedrate through the M220 gcode command
 
     def moveAbsolute(self, axis, pos, feedrate):
-        self.SendGCode('M82') # change to absolute positioning
-        self.SendGCode('G1 {}{} F{}'.format(axis, str(pos), str(feedrate))) # move the specified axis at the set feedrate
+        self.sendGCode('M82') # change to absolute positioning
+        self.sendGCode('G1 {}{} F{}'.format(axis, str(pos), str(feedrate))) # move the specified axis at the set feedrate
 
     def save_settings(self):
-        self.SendGCode('SAVE_CONFIG') # save the current configuration changes
+        self.sendGCode('SAVE_CONFIG') # save the current configuration changes
 
     def setExtTemp(self, target, toolnum=0):
         self.sendGCode("M104 T%s S%s" % (toolnum, str(target)))
 
     def setExtTemp(self, target):
-        self.SendGCode('M104 S' + str(target))
+        self.sendGCode('M104 S' + str(target))
 
     def setZOffset(self, offset):
-        self.SendGCode('SET_GCODE_OFFSET Z={} MOVE=1'.format(str(offset)))
+        self.sendGCode('SET_GCODE_OFFSET Z={} MOVE=1'.format(str(offset)))
 
     def add_mm(self, axis, zoffset):
-        self.SendGCode('TESTZ Z=' + str(zoffset))
+        self.sendGCode('TESTZ Z=' + str(zoffset))
 
     def log(self, msg, *args, **kwargs):
         if self._logging:
-- 
2.39.5 (Apple Git-154)

